src/browser/connection/gateway.js

```diff
@@ -63,6 +62,18 @@ export default class BrowserConnectionGateway {
         return true;
     }
 
+    static _fetchRequestData (req, callback) {
+        let data = '';
+
+        req.on('data', chunk => {
```

Comment:
(maybe out-of-scope) Missing `error` handler. If you don't plan to fix it right now, please create a ticket and add a `// TODO` comment.

Reply:
The current code doesn't handle the 'error' event.
```js
   static onInitScriptResponse (req, res, connection) {
        if (BrowserConnectionGateway.ensureConnectionReady(res, connection)) {
            let data = '';

            req.on('data', chunk => {
                data += chunk;
            });

            req.on('end', () => {
                connection.handleInitScriptResult(data);

                res.end();
            });
        }
    }
```

I will create a separate issue for this case.

Reply:
https://github.com/DevExpress/testcafe/issues/4563